### DISCUSSION TITLE
Add support for py::issubclass

### DISCUSSION THREAD

**Discussion by markuspi** [pybind/pybind11#3627] [Ideas]
pybind11 is missing a `py::issubclass` functionality, analogous to `py::isinstance`.

I am thinking of adding something like this to pytypes.h:
```c++
template <typename T>
bool issubclass(handle cls) {
    const auto parent = type::handle_of<T>();
    return issubclass(cls, parent);
}

inline bool issubclass(handle cls, handle parent) {
    const auto result = PyObject_IsSubclass(cls.ptr(), parent.ptr());
    if (result == -1)
        throw error_already_set();
    return result != 0;
}
```
However, I am not sure this will work for subclasses of `object`, as they are also handled seperately in `py::isinstance`.
I would appreciate your feedback!

**Comment by damienmarchal** [comment]
Is there any updates on this conversation ? 
I may have some use for such feature.
